apis/kueue/v1alpha1/workload_webhook.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	defaultPodSetName = "main"
+)
+
+// log is for logging in this package.
+var workloadlog = logf.Log.WithName("workload-resource")
+
+func (r *Workload) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// +kubebuilder:webhook:path=/mutate-kueue-x-k8s-io-v1alpha1-workload,mutating=true,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=workloads,verbs=create;update,versions=v1alpha1,name=mworkload.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Workload{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Workload) Default() {
+	workloadlog.Info("default", "name", r.Name)
+
+	for i := range r.Spec.PodSets {
+		podSet := &r.Spec.PodSets[i]
+		if len(podSet.Name) == 0 {
+			podSet.Name = defaultPodSetName
+		}
+	}
+
+	if r.Spec.Admission != nil {
+		for i := range r.Spec.Admission.PodSetFlavors {
+			flavor := &r.Spec.Admission.PodSetFlavors[i]
+			if len(flavor.Name) == 0 {
+				flavor.Name = defaultPodSetName
+			}
+		}
+	}
```

Comment:
remove lines 56-63; defaulting is not applicable here. This field is set by the scheduler.